enhancements/monitoring/alerts-ui-management.md

```diff
@@ -0,0 +1,642 @@
+| GET    | `/rules/{ruleId}`                  | Fetch a single rule’s full definition and relabel context. | **Required** — merges `AlertRelabelConfig` with platform alert.|
+| PATCH  | `/rules/{ruleId}`                  | Update a single rule (platform or user‑defined). | |
+| DELETE | `/rules/{ruleId}`                  | Delete a user‑defined rule. | |
+| POST   | `/rules`                           | Bulk update rules (e.g., severity/status/labels).  | |
+| DELETE | `/rules`                           | Bulk delete user‑defined rules. | |
+
+Filter parameters for `/rules`:
+
+- `name`
+- `group`
+- `namespace`
+- `component`
+- `severity`
+- `state` (one of: `enabled`, `disabled`) — rule enablement status. Note: “silenced” is not a rule state.
+- `has_silenced_alerts` (boolean) - filter rules that currently have one or more associated alerts silenced. - TBD
+- `source`
+- `notification_receiver`
+- Other alert labels (arbitrary `labelKey=labelValue` pairs)
+
+**Health**
+
+| Method | Path        | Description                 | Notes |
+|--------|-------------|-----------------------------|-------|
+| GET    | `/health`   | Basic health‑check endpoint |       |
+
+
+
+### Data Model
+- Store user alerts in `PrometheusRule`s in the specified project.
+- Update platform alerts by writing relabel rules into the `AlertRelabelConfig` CR in `cluster-monitoring-config`.
+- Use labels for disable/enable flags; rely on existing CRDs - Applies only for Platform alert rules.
+
+### Migration
+- Automated tool to detect existing `PrometheusRule` CRDs and import them into the new system
+- Preserve existing labels, annotations and support rollback on errors
+
+### GitOps / Argo CD Compliance
+- All generated `PrometheusRule` and `AlertRelabelConfig` resources remain declarative and suitable for committing to Git
+
+### Workflow Description
+
+## Pain Points Addressed by this Design
+- **Generic and Non-Actionable Alerts:** The design will provide more actions from an alert and link to resources like logs and Korrel8r for troubleshooting.
+- **Alert Noise and Data Overload:** Grouping, advanced filters, and saved filters will help reduce noise and the need for repetitive filtering.
+- **Missed Alarms or Missing Data:** Users will be able to create flexible alert definitions directly in the UI to monitor any data type, configure notifications, and link a runbook.
+
+## Pain Points Not Directly Addressed
+- High Resource Consumption of Observability Tools
+- Lack of Unified Views (within a single-cluster scope)
+
+## Risks & Mitigations
+- **Performance**: split large rule sets into multiple CRs; benchmark reconciliation latency
+- **Misconfiguration**: validate syntax; prevent disabling protected base alerts via exclusion list
+- **Complexity**: sensible defaults; UI guidance; limit advanced options in MVP
+
+## Graduation Criteria
+- **Tech Preview**: basic CRUD, clone, disable/enable, silencing via UI/API; migration tool v1
```

Comment:
How would users enable the feature when tech-preview? Feaure gates work well for new CRDs, less for features which aren't tied to CRDs.